Makefile

```diff
@@ -126,12 +126,36 @@ vet: generate ## Run go vet against code.
 test: test-unit test-integration ## Run all tests (unit and integration).
 
 .PHONY: test-unit
-test-unit: generate ## Run unit tests only.
-	go test ./... -v -timeout 30s
+test-unit: generate ## Run unit tests with coverage.
+	@mkdir -p coverage
+	go test ./... -v -timeout 30s -coverprofile=coverage/coverage-unit.out -covermode=atomic
+	@echo "Unit test coverage saved to coverage/coverage-unit.out"
 
 .PHONY: test-integration
-test-integration: generate manifests build-ebpf ## Run integration tests.
-	EBPF_BUILD_DIR=$(EBPF_BUILD_DIR) ANTIMETAL_BPF_PATH=$(EBPF_BUILD_DIR) go test -tags integration ./... -v -timeout 60s
+test-integration: generate manifests build-ebpf ## Run integration tests with coverage.
+	@mkdir -p coverage
+	EBPF_BUILD_DIR=$(EBPF_BUILD_DIR) ANTIMETAL_BPF_PATH=$(EBPF_BUILD_DIR) go test -tags integration ./... -v -timeout 60s -coverprofile=coverage/coverage-integration.out -covermode=atomic
+	@echo "Integration test coverage saved to coverage/coverage-integration.out"
+
+.PHONY: test-coverage
+test-coverage: test ## Run all tests and generate coverage report.
+	@make coverage-report
+
+.PHONY: coverage-merge
+coverage-merge: ## Merge unit and integration coverage files.
+	@.github/scripts/merge-coverage.sh
+
+.PHONY: coverage-html
+coverage-html: coverage-merge ## Generate HTML coverage report.
+	@.github/scripts/generate-coverage-html.sh
+
+.PHONY: coverage-report
+coverage-report: coverage-merge ## Display coverage summary with threshold check.
+	@.github/scripts/coverage-report.sh || true
+
+.PHONY: coverage-check
+coverage-check: coverage-merge ## Check coverage meets threshold (fails if below).
+	@.github/scripts/coverage-report.sh
```

Comment:
I don't see these make targets used in the github actions workflow. Should either get rid of this and the scripts or have both the github actions workflow and test-coverage-lima.sh use these make targets so that the coverage generation is consistent